### ISSUE TITLE
Make muzzle reference creation package(s) configurable

### ISSUE THREAD

**Issue by pavolloffay** [open-telemetry/opentelemetry-java-instrumentation#1395]
The `instrumentationMuzzle` is created only for instrumenters in `io.opentelemetry.instrumentation.`

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/dadaac9a8cba35e3054fea7e62996e9b6c3e1693/javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/ReferenceCreationPredicate.java#L15

In my custom java agent build I am adding instrumentations in a different package and these instrumentations should have also `instrumentationMuzzle` properly created.

The package prefixes could be configured via a system property which would be set at build time. I don't think we need an SPI for it.

**Comment by mateuszrzeszutek** [comment]
It needs to be available both in compile-time and runtime - `ReferenceMatcher` is also using this predicate.

Are these the same packages as in `BootstrapPackagePrefixesHolder`?

By the way, I'm thinking that maybe we need to have some sort of `AgentInternalConfig` class that groups all such properties together, and a single SPI for changing that config (similar to `TracerCustomizer`). This way a vendor distribution could configure all its needs in a single place.

**Comment by pavolloffay** [comment]
> Are these the same packages as in BootstrapPackagePrefixesHolder?

no, the packages in `BootstrapPackagePrefixesHolder` are different (not instrumentation) - these are classes that provide common functionality for instrumentations.

**Comment by mateuszrzeszutek** [comment]
Another idea based on: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2524#discussion_r589388031

Right now if we want to include a 3rd party library instrumentation in the javaagent (e.g. [couchbase 3.1 instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2524)) we're forced to add all classes used in the library instrumentation as `additionalHelperClassNames()`.
That's because muzzle collects neither helper classes nor references (used symbols) from 3rd party packages/classes.

To fix that a new customization point should be introduced to the `InstrumentationModule`: a method that returns a `Predicate<String>` (or similar) that returns a predicate determining whether a passed class name refers to an instrumentation helper class. The default behavior would be the current one e.g. `InstrumentationClassPredicate`.

To sum it up, instead of using SPI (as proposed above) we should allow customizing this behavior per instrumentation module.
